swim/heal_partition.go

```diff
@@ -69,22 +69,46 @@ func nodesThatNeedToReincarnate(MA, MB []Change) (changesForA, changesForB []Cha
 			continue
 		}
 
+		// TODO: DRY
```

Comment:
I think if we want to refactor the two blocks below, the best time to do it is now. Or agree we will not, and remove this comment.

I think it's good as-is.